packages/mini-oxygen/src/vite/entry-error.ts

```diff
@@ -0,0 +1,113 @@
+import type {ServerResponse} from 'node:http';
+import {readFile} from 'node:fs/promises';
+import type {ViteDevServer} from 'vite';
+import {getErrorPage} from '../common/error-page.js';
+import type {Response} from '../worker/index.js';
+
+export function isEntrypointError(webResponse: Response) {
+  return (
+    webResponse.status === 503 &&
+    webResponse.statusText === 'executeEntrypoint error'
+  );
+}
+
+export type CustomEntryPointErrorHandler = (params: {
+  optimizableDependency?: string;
+  stack: string;
+}) => void | Promise<{
+  status?: number;
+  body?: string;
+  headers?: Record<string, string>;
+} | void>;
+
+export async function handleEntrypointError(
+  viteDevServer: ViteDevServer,
+  webResponse: Response,
+  res: ServerResponse,
+  entryPointErrorHandler?: CustomEntryPointErrorHandler,
+) {
+  const stack = (await webResponse.text())
+    .split('\n')
+    .filter((line) => !line.includes('mini-oxygen'))
+    .join('\n');
+
+  const optimizableDependency = await findOptimizableDependency(
```

Comment:
Seems like the logic inside `findOptimizableDependency` is possible to throw in some situations. If it does, it will end up in the outer catch that logs`'MiniOxygen: Error during evaluation:`. That has the potential to confuse the developer. I'd rather see the original error from vite, than an error in our logic trying to optimize the dependencies. Maybe we should try catch around this, and throw the original `entryPointErrorHandler` if it fails?

Reply:
In what situation can it throw? It already has a `.catch()` in the only async call (readFile) and the rest shouldn't create any issue 🤔 

I can catch it here as well I guess to be extra safe.